daemon/api.go

```diff
@@ -457,13 +464,51 @@ func getStore(c *Command) snapstate.StoreService {
 	return snapstate.Store(st)
 }
 
+func getSections(c *Command, r *http.Request, user *auth.UserState) Response {
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("cannot find route for snaps")
+	}
+
+	theStore := getStore(c)
+
+	sections, err := theStore.Sections(user)
+	switch err {
+	case nil:
+		// pass
+	case store.ErrEmptyQuery, store.ErrBadQuery:
+		return BadRequest("%v", err)
+	case store.ErrUnauthenticated:
+		return Unauthorized(err.Error())
```

Comment:
why `("%v", err)` in the others and `err.Error()` here? AFAIK they're both the same unless `err==nil`, which you already split out explicitly